ls1intum/Artemis
# Fix an issue in the programming exercise details page for instructors

Hibernate in combination with Hazelcast randomly leads to Null Pointer Exceptions when fetching template or solution participations for programming exercises in the `canAccessParticiption(...)` method. This happens more often on multi node sever setups, probably because of caching issues related to Hazelcast.

I changed how the programming exercise details page for instructors is loaded, which should solve the issue. The page needs a larger redesign with respect to the number of REST calls, but the change should solve the issue that you typically notice on Staging and Production, but not necessarily on the simple test servers (TS1, TS2, TS3, TS5) because they only have one server node.

I also removed a too similar method and unified the approach for `canAccessParticiption(...)`

### Checklist
- [x] Server: I added server integration tests 

### Steps for Testing
0. Execute the following steps on the staging environment: https://artemis-staging.ase.in.tum.de
1. Open the programming exercise details page of a programming exercise and make sure the template and solution results (at the bottom) are displayed correctly
2. Open `Edit in Editor` and submit multiple times for template and solution, make sure the result changes
3. Regularly check if the programming exercise details page then displays the correct, i.e. most recent, result for template and solution without any error messages (after the build has taken place)
